test/unit/logic/delegate.js

```diff
@@ -370,6 +326,38 @@ describe('delegate', function () {
 				});
 			});
 		});
+
+		describe('when transaction is valid', function () {
```

Comment:
This `describe` block has the same description as another `describe` block at the same level. In my previous review the suggestion was to drop this whole section one level deeper but keep it in the same position in the file. The new nested block should be given a description corresponding to what the `beforeEach` hook sets up, i.e. what situation is represented by `delegate.checkConfirmed` calling its callback with`null`?